biogeochem/EDPatchDynamicsMod.F90

```diff
@@ -1019,411 +1127,697 @@ subroutine set_patchno( currentSite )
-                     understorey_dead*(struct_c + sapw_c)  
-                canopy_mortality_leaf_litter(p)= canopy_mortality_leaf_litter(p)+ &
-                     understorey_dead*leaf_c
-                canopy_mortality_root_litter(p)= canopy_mortality_root_litter(p)+ &
-                      understorey_dead*(fnrt_c + store_c)
-                
-                if( hlm_use_planthydro == itrue ) then
-                   call AccumulateMortalityWaterStorage(currentSite,currentCohort, understorey_dead)
-                end if
-
-             ! FIX(SPM,040114) - clarify this comment
-             ! grass is not killed by canopy mortality disturbance events.
-             ! Just move it into the new patch area. 
-             else 
-                ! no-op
-             endif
-          endif
+             ! Transfer wood of dying trees to AG CWD pools
+             new_litt%ag_cwd(c) = new_litt%ag_cwd(c) + ag_wood * &
+                    SF_val_CWD_frac(c) * donate_m2
 
+             curr_litt%ag_cwd(c) = curr_litt%ag_cwd(c) + ag_wood * &
+                   SF_val_CWD_frac(c) * retain_m2
+             
+             ! Transfer wood of dying trees to BG CWD pools
+             do sl = 1,currentSite%nlevsoil
+                new_litt%bg_cwd(c,sl) = new_litt%bg_cwd(c,sl) + bg_wood * &
+                       currentSite%rootfrac_scr(sl) * SF_val_CWD_frac(c) * &
+                       donate_m2
+
+                curr_litt%bg_cwd(c,sl) = curr_litt%bg_cwd(c,sl) + bg_wood * &
+                      currentSite%rootfrac_scr(sl) * SF_val_CWD_frac(c) * &
+                      retain_m2
+             end do
+          end do
+
+          ! Transfer fine roots of dying trees to below ground litter pools
+          do dcmpy=1,ndcmpy
+              dcmpy_frac = GetDecompyFrac(pft,fnrt_organ,dcmpy)
+              do sl=1,currentSite%nlevsoil
+                  new_litt%root_fines(dcmpy,sl) = new_litt%root_fines(dcmpy,sl) + &
+                        num_dead * currentSite%rootfrac_scr(sl) * &
+                        (fnrt_m + store_m*(1.0_r8-EDPftvarcon_inst%allom_frbstor_repro(pft))) * &
+                        donate_m2 * dcmpy_frac
+                  
+                  curr_litt%root_fines(dcmpy,sl) = curr_litt%root_fines(dcmpy,sl) + &
+                        num_dead * currentSite%rootfrac_scr(sl) * &
+                        (fnrt_m + store_m*(1.0_r8-EDPftvarcon_inst%allom_frbstor_repro(pft))) * &
+                        retain_m2 * dcmpy_frac
+              end do
+          end do
+              
+          ! Transfer some of the storage that is shunted to reproduction
+          ! upon death, to the seed-pool. This is was designed for grasses,
+          ! but it is possible that some trees may utilize this behavior too
+
+          seed_mass = num_dead * store_m * EDPftvarcon_inst%allom_frbstor_repro(pft)
+
+          ! SEED DISTRIBUTION IS BREAKING MASS CONSERVATION RIGHT NOW...
```

Comment:
is this still the case and if so do we need to keep track of it with an issue?

Reply:
yes, I was not able to get tight mass conservation on this, I will make an issue of this once we pull.  
Note that we do allow storage mass to convert to seed in certain grasses, it just doesn't disperse and mix at the site level.